### PR TITLE
feature: Separate API and dashboard

### PR DESCRIPTION
**Pull request by kylecarbs**
Adds routing for separating cemanager and the dashboard.

Related: https://github.com/cdr/m/pull/6315

### CODE REVIEW

File: templates/dashboard.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                - curl
+                - -s
+                - localhost:8080/healthz
+            initialDelaySeconds: 30
+            failureThreshold: 14
+            # Poll envproxy every 20 seconds. If we fail 12 times (4 minutes)
+            # the manager is most likely in a bad spot. It's possible for
+            # migrations to take extended periods of time, so we need to
+            # account for that.
+            periodSeconds: 20
```
Comment (cmoog) [inline templates/dashboard.yaml:59]: looks like a holdover from cemanager/envproxy charts

File: templates/dashboard.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    matchLabels:
+      coder.deployment: dashboard
+  template:
+    metadata:
+      labels:
+        coder.deployment: dashboard
+    spec:
+      restartPolicy: Always
+      serviceAccountName: dashboard
+      containers:
```
Comment (cmoog) [inline templates/dashboard.yaml:25]: This needs resource requests

**Review by cmoog — CHANGES_REQUESTED** [review]

**Review by nathanpotter — APPROVED** [review]
LGTM

File: templates/dashboard.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            periodSeconds: 20
+          livenessProbe:
+            exec:
+              command:
+                - curl
+                - -s
+                - localhost:3000/healthz
+            initialDelaySeconds: 30
+            failureThreshold: 14
+            periodSeconds: 20
```
Comment (cmoog) [inline templates/dashboard.yaml:59]: These should probably be more aggressive for the dashboard

**Review by cmoog — APPROVED** [review]

**Pull request merged**